timamero/react-nodejs-video-chat
# Refactor private chat

### Description
This PR updates the private chat so that the video stream between the two peers start when they enter the room. The text chat is updated so that it can be toggled.

In the backend, the socket event handlers in pubsub/privateChat.ts were updated so that the peer video connection started when the users enter the room.

### Type

### Demo

https://user-images.githubusercontent.com/54559570/157052759-651c9bb6-7734-4320-afdc-40ca0688bcb3.mp4

### Testing
- Clone this branch and install dependencies
- If your MongoDB database is local make sure that it is running
- Run the development server `npm run dev`
- Open two Firefox browsers
- Go the page localhost:3000 in each browsers
- Initiate a private chat by clicking `Invite to Chat` in one of the browsers

Notes:
- May not work with Chrome
- Can work with one or two webcams.  If there is only one webcam, the local and remote feeds will be the same.
